monkey/tests/unit_tests/common/event_queue/test_pypubsub_event_queue.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass
+
+import pytest
+from pubsub import pub
+
+from common.event_queue.pypubsub_event_queue import (
+    INTERNAL_ALL_EVENT_TYPES_TOPIC,
+    PyPubSubEventQueue,
+)
+from common.events import AbstractEvent
+
+EVENT_TAG_1 = "event tag 1"
+EVENT_TAG_2 = "event tag 2"
+
+
+@dataclass(frozen=True)
+class EventType(AbstractEvent):
+    source = "1234"
+    target = None
+    timestamp = 0.0
+    tags = [EVENT_TAG_1, EVENT_TAG_2]
+
+
+pypubsub_event_queue = PyPubSubEventQueue(pub)
+
+
+@pytest.fixture()
+def subscriber_1_calls():
+    return []
+
+
+@pytest.fixture()
+def subscriber_2_calls():
+    return []
+
+
+@pytest.fixture
+def subscriber_1(subscriber_1_calls):
+    def fn(event, topic=pub.AUTO_TOPIC):
+        subscriber_1_calls.append(topic.getName())
+
+    return fn
+
+
+@pytest.fixture
+def subscriber_2(subscriber_2_calls):
+    def fn(event, topic=pub.AUTO_TOPIC):
+        subscriber_2_calls.append(topic.getName())
+
+    return fn
+
+
+@pytest.mark.usefixtures("subscriber_1", "subscriber_2", "subscriber_1_calls", "subscriber_2_calls")
+def test_topic_subscription(subscriber_1, subscriber_2, subscriber_1_calls, subscriber_2_calls):
+    pypubsub_event_queue.subscribe_type(EventType, subscriber_1)
+    pypubsub_event_queue.subscribe_tag(EVENT_TAG_2, subscriber_1)
+    pypubsub_event_queue.subscribe_tag(EVENT_TAG_1, subscriber_2)
+    pypubsub_event_queue.publish(EventType)
+
+    assert subscriber_1_calls == [EventType.__name__, EVENT_TAG_2]
+    assert subscriber_2_calls == [EVENT_TAG_1]
+
+
+def test_subscribe_all():
+    subscriber_calls = []
+
+    def subscriber(topic=pub.AUTO_TOPIC):
+        subscriber_calls.append(topic.getName())
+
+    pypubsub_event_queue.subscribe_all(subscriber)
+    pypubsub_event_queue.publish(EventType)
+
+    assert subscriber_calls == [
+        EventType.__name__,
+        INTERNAL_ALL_EVENT_TYPES_TOPIC,
+        EVENT_TAG_1,
+        EVENT_TAG_2,
+    ]
```

Comment:
No! If I subscribe for all events, I should get each event only once. See https://github.com/guardicore/monkey/pull/2182#discussion_r941230712

Reply:
To clarify, "subscribe all" means "all events", not "all topics". The `IEventQueue` doesn't expose the concept of a topic.

Feel free to change the name of the method to `subscribe_all_events()` if you think that helps clarify its purpose.

Reply:
There's another `subscribe_all_event_types()` method in PyPubSubEventQueue for that. I'm not sure if we should completely remove the `subscribe_all()` method.

Added another UT for `subscribe_all_event_types()`.

Reply:
`subscribe_all_event_types()` is not specified in `IEventQueue`. I've removed it from `PyPubSubEventQueue` and renamed `subscribe_all()` to `subscribe_all_events()`.